lib/google_maps.ex

```diff
@@ -1523,6 +1523,72 @@ defmodule GoogleMaps do
+    request, photo references will be returned for relevant photographic content.
+    The Nearby Search and Text Search requests also return a single photo reference
+    per place, when relevant. Using the Photo service you can then access the
+    referenced photos and resize the image to the optimal size for your application.
+
+  ## Args:
+  * `photo_reference` — A string identifier that uniquely identifies a photo.
+    Photo references are returned from either a Place Search or Place Details request.
+
+  * `dimension_constraint` — A tuple of the form `{:maxheight, integer}` or
+    `{:maxwidth, integer}`. Specifies the maximum desired height or width, in pixels,
+    of the image returned by the Place Photos service. If the image is smaller than the
+    values specified, the original image will be returned. If the image is larger in
+    either dimension, it will be scaled to match the smaller of the two dimensions,
+    restricted to its original aspect ratio. Both the maxheight and maxwidth properties
+    accept an integer between 1 and 1600.
+
+  ## Options:
+    While no optional parameters for Place Photos exist at the time of writing, the
+    options keyword list supports the `key` parameter in case an explicit API key
+    is to be supplied.
+
+  ## Returns
+
+    This function returns `{:ok, body}` if the request is successful.
+    The returned body is a binary containing jpeg image data.
+
+    If the API key is invalid or the request has exceeded your quota, the function
+    returns `{:error, 403}`.
+
+    Otherwise, if the request is invalid, the return value will be `{:error, 400}`.
+    This likely indicates an incorrect photo reference, but can also mean an
+    invalid value for `maxwidth` or `maxheight`.
+
+  ## Examples
+
+      # Request with an invalid API key
+      iex> photo_reference = "CnRtAAAATLZNl354RwP_9UKbQ_5Psy40texXePv4oAlgP4qNEkdIrkyse7rPXYGd9D_Uj1rVsQdWT4oRz4QrYAJNpFX7rzqqMlZw2h2E2y5IKMUZ7ouD_SlcHxYq1yL4KbKUv3qtWgTK0A6QbGh87GB3sscrHRIQiG2RrmU_jF4tENr9wGS_YxoUSSDrYjWmrNfeEHSGSc3FyhNLlBU"
+      iex> GoogleMaps.place_photo(photo_reference, {:maxwidth, 500}, key: "invalid key")
+      {:error, 403}
+
+      # Request with an invalid photo reference
+      iex> GoogleMaps.place_photo("invalid reference", {:maxwidth, 500})
+      {:error, 400}
+
+      # Request a photo with a maximum height of 100
+      iex> photo_reference = "CnRtAAAATLZNl354RwP_9UKbQ_5Psy40texXePv4oAlgP4qNEkdIrkyse7rPXYGd9D_Uj1rVsQdWT4oRz4QrYAJNpFX7rzqqMlZw2h2E2y5IKMUZ7ouD_SlcHxYq1yL4KbKUv3qtWgTK0A6QbGh87GB3sscrHRIQiG2RrmU_jF4tENr9wGS_YxoUSSDrYjWmrNfeEHSGSc3FyhNLlBU"
+      iex> {:ok, response} = GoogleMaps.place_photo(photo_reference, {:maxheight, 100})
+      iex> is_binary(response)
+      true
+  """
+  @spec place_photo(String.t, {:maxwidth, integer()} | {:maxheight, integer()}, options()) :: Response.t()
+  def place_photo(photo_reference, {constraint, value} = dimension_constraint, options \\ [])
```

Comment:
I have tried a request with both `maxwidth` and `maxheight` parameters and Google does support that.

What do you think about changing the function signature to take both parameters, and set default to missing one. For example:

```elixir
def place_photo(photo_reference, max_width, max_height)
def place_photo(photo_reference, max_width), do: place_photo(photo_reference, max_width, nil)
def place_photo(photo_reference, nil, max_height)
```

The reason is that we would like to keep the signature of the functions in this module similar, and required parameters should be part of function arity.

Reply:
Good catch!
I do think we should keep the `opts` at the end though, for consistency with the other APIs such as being able to specify a `key` for a single call. But then it might get a little bit hairy:

```elixir
def place_photo(photo_reference, max_width, max_height, opts)
def place_photo(photo_reference, max_width, max_height) when is_integer(max_height), do:
place_photo(photo_reference, max_width, max_height, [])
def place_photo(photo_reference, max_width, opts) when is_list(opts), do: 
place_photo(photo_reference, max_width, nil, opts)
def place_photo(photo_reference, max_width), do:
place_photo(photo_reference, max_width, nil, [])
```

:thinking:

Reply:
Agreed. We definitely want to keep `opts` at the end. We just need to work out the function clauses order to handle all the cases. I'll give it some more thought as well.

Reply:
Sorry for the lateness.

I have tried with the following combinations and it seems to work:

```elixir
def place_photo(photo_reference, max_width) do
  place_photo(photo_reference, max_width, 1600, [])
end

def place_photo(photo_reference, max_width, max_height) do
  place_photo(photo_reference, max_width, max_height, [])
end

def place_photo(photo_reference, max_width, max_height, opts)
when is_binary(photo_reference)
and is_integer(max_width) and max_width >= 1 and max_width <= 1600
and is_integer(max_height) and max_height >=1 and max_height <= 1600
and is_list(opts)
do
  # Perform request.
end
```

If the user does not care about `max_width`, they have to put 1600 in, unless there is another clause to handle `nil` value.